include/net/lwm2m.h

```diff
@@ -1000,21 +1020,57 @@ int lwm2m_engine_register_delete_callback(uint16_t obj_id,
 int lwm2m_engine_set_res_data(const char *pathstr, void *data_ptr, uint16_t data_len,
```

Comment:
We should mark the function with `__deprecated` keyword, so that it produces a compiler warning.

Reply:
Marked. Thanks.